README.md

```diff
@@ -17,11 +38,107 @@ Corpus][corpus], as described by Thorsten Brants and Alex Franz, and [distribute
 Linguistic Data Consortium. Note that this data **"may only be used for linguistic
 education and research"**, so for any other usage you should acquire a different data set.
 
-For the microbenchmark included in this repository, instant-segment is ~17x faster than
+For the microbenchmark included in this repository, Instant Segment is ~17x faster than
 the Python implementation. Further optimizations are planned -- see the [issues][issues].
 The API has been carefully constructed so that multiple segmentations can share
 the underlying state to allow parallel usage.
 
+## Installing
+
+### Python **(>= 3.9)**
+
+```sh
+pip install instant-segment
+```
+
+### Rust
+
+```toml
+[dependencies]
+instant-segment = "*"
+```
+
+## Using
+
+Instant Segment works by segmenting a string into words by selecting the splits with the highest probability given a corpus of words and their occurances.
+
+For instance, provided that `choose` and `spain` occur more frequently than `chooses` and `pain`, Instant Segment can help you split the string `choosespain.com` into [`ChooseSpain.com`](https://instantdomainsearch.com/search/sale?q=choosespain) which more likely matches user intent.
+
+```python
+import instant_segment
+
+
+def main():
+    unigrams = []
+    unigrams.append(("choose", 50))
+    unigrams.append(("chooses", 10))
+    unigrams.append(("spain", 50))
+    unigrams.append(("pain", 10))
+
+    bigrams = []
+    bigrams.append((("choose", "spain"), 10))
+    bigrams.append((("chooses", "pain"), 10))
```

Comment:
Not sure about having this code in the README. It makes it look quite verbose for what is a very contrived example. Also don't think we need this in two languages, since I think the earlier example established how closely both APIs are related?

Reply:
I'll move these to example files. I do like showing both languages since I want to make it as clear as possible to newcomers that the API is available in both Python and Rust.